lib/HeroBonus.cpp

```diff
@@ -1283,6 +1316,8 @@ JsonNode Bonus::toJsonNode() const
 		root["val"].Integer() = val;
 	if(valType != ADDITIVE_VALUE)
 		root["valueType"].String() = vstd::findKey(bonusValueMap, valType);
+	if(stacking != "")
+		root["stacking"].String() = stacking;

```

Comment:
Some fields are missing here (f.e. source and source id). Is that intentional?

Reply:
Yes - this function is being used to convert bonuses to a format you would store in a config file - here source and source_id wouldn't make an appearance here.

Reply:
But what about `duration`, `turnsRemain`, `description`?

Reply:
Not `turnsRemain` I think, as that is dynamic, but `duration` and `description` would make sense. Added this functions when converting old bonus formats into new, currently it's just used for creating log outputs that can be copy-pasted if needed.

Reply:
Actually, it looks like there is duplicated functionality with `JsonUtils::unparseBonus`, though it seems that the latter is never used anywhere.

Reply:
turnsRemain is dynamic but configuration value is used as initial value for temporary bonuses (f.e. for spells it by default set to spellpower, but could be explicitly defined too)

Reply:
fixed / cleaned up